blog/content/post/2020-02-13-rpi4_yocto.md

```diff
@@ -0,0 +1,272 @@
+`Yocto/meta-raspberrypi/wic` we can see the following lines:
+
+```shell
+# short-description: Create Raspberry Pi SD card image
+# long-description: Creates a partitioned SD card image for use with
+# Raspberry Pi. Boot files are located in the first vfat partition.
+
+part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
+part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label root --align 4096
+```
+
+If we add this to our project, an image will automatically create required
+partitions for boot and root on SD card after flashing. To provide that
+functionality everything we have to do is to append two lines to the local.conf
+
+```shell
+WKS_FILE = "sdimage-raspberrypi.wks"
+```
+
+This will tells Yocto to add `sdimage-raspberrypi.wks` to the project. And the
+second line which is responsible for creating two compressed output files.
+
+```shell
+IMAGE_FSTYPES += "wic.gz wic.bmap"
+```
+
+To save some time in the future we can also make Yocto responsible for enabling
+uart what we will need in the future, so append to local.conf following line:
+
+```shell
+ENABLE_UART = "1"
+```
+
+That's it. Now we can use Bitbake to build the image.
+
+```shell
+cd ~/Yocto/poky
+source oe-init-build-env
+bitbake core-image-minimal
+```
+
+If everything was succesful w should see message simillar to this:
+
+```shell
+Parsing recipes: 100% |######################################################################################################################################################################| Time: 0:00:27
+Parsing of 801 .bb files complete (0 cached, 801 parsed). 1327 targets, 63 skipped, 0 masked, 0 errors.
+NOTE: Resolving any missing task queue dependencies
+
+Build Configuration:
+BB_VERSION           = "1.44.0"
+BUILD_SYS            = "x86_64-linux"
+NATIVELSBSTRING      = "universal"
+TARGET_SYS           = "aarch64-poky-linux"
+MACHINE              = "raspberrypi4-64"
+DISTRO               = "poky"
+DISTRO_VERSION       = "3.0.2"
+TUNE_FEATURES        = "aarch64 cortexa72 crc crypto"
+TARGET_FPU           = ""
+meta                 
+meta-poky            
+meta-yocto-bsp       = "zeus:73fe0e273b4e00dcb08122c4f54fc65316e2a793"
+meta-raspberrypi     = "zeus:d17588fe8673b794b589335a753f4c1c90e12f88"
+
+Initialising tasks: 100% |###################################################################################################################################################################| Time: 0:00:01
+Sstate summary: Wanted 4 Found 0 Missed 4 Current 828 (0% match, 99% complete)
+NOTE: Executing Tasks
+NOTE: Setscene tasks completed
+NOTE: Tasks Summary: Attempted 2672 tasks of which 2617 didn't need to be rerun
+and all succeeded.
+```
+
+Now among many files located in `/Yocto/poky/build/tmp/deploy/images/raspberrypi4-64`
+we can find the result of our build. The easiest way to do that is to type:
+
+```shell
+find . -print | grep 'wic'
```

Comment:
Awkward way to do that. Simple ls *wic* or print . -name '*wic*' will do the same I believe